src/langcheck/metrics/en/pairwise_text_quality.py

```diff
@@ -12,6 +16,100 @@
+    prompt_params: List[dict[str, str | None]],
+    eval_model: EvalClient,
+    k: int = 5,
+    n: int = 5,
+    seed: int | None = None,
+) -> List[float | None]:
+    """Compute a confidence score for the pairwise comparison metric based on
+    the method Simulated Annotators proposed in the paper "Trust or Escalate:
+    LLM Judges with Provable Guarantees for Human Agreement"
+    (https://arxiv.org/abs/2407.18370)
+
+    Args:
+        prompt_params: The parameters used to populate the prompt template.
+        eval_model: The EvalClient instance used for the evaluation.
+        k: the number of examples of preference annotations
+        n: the numbre of simulated annotators
+        seed: the random seed for selecting the few-shot examples
+    Returns:
+        A confidence score for the pairwise comparison metric
+    """
+    # Load preprocessed chatarena data
+    cwd = Path(__file__).parent
+    with open(cwd / "processed_chatarena_examples.jsonl") as f:
+        chatarena_data = [json.loads(line) for line in f]
+    assert len(chatarena_data) >= k, "Not enough examples in the chatarena data"
+
+    if seed is not None:
+        random.seed(seed)
+
+    # Load the prompt template
+    prompt_template = get_template(
+        "en/confidence_estimating/simulated_annotators.j2"
+    )
+
+    confidence_scores = []
+    for prompt_param in prompt_params:
+        # Simulate n annotators
+        prompts = []
+        for _ in range(n):
+            # Generate few-shot examples
+            few_shot_examples = random.sample(chatarena_data, k)
+
+            # Construct the full prompt using k few-shot examples
+            prompt_param["few_shot_examples"] = "\n".join(
+                f"[Question]\n{example['prompt']}\n\n"
+                "[Assistant A's response]\n{example['model_a']}\n\n"
+                "[Assistant B's response]\n{example['model_b']}\n\n"
+                "[Verdict]\n{example['winner']}\n"
+                for example in few_shot_examples
+            )
+            prompts.append(prompt_template.render(prompt_param))
+
+        # Get the response and top five logprobs of the first token
+        responses: List[Optional[ResponseDict]] = (
+            eval_model.get_text_responses_with_log_likelihood(
+                prompts, top_logprobs=5
+            )
+        )
+        scores_a, scores_b = [], []
+        for response in responses:
+            if response:
+                response = cast(ResponseDict, response)
+                response_first_token = cast(
+                    LogProbInfo, response["response_logprobs"][0]
+                )
+                top_five_logprobs = cast(
+                    List[TokenLogProb],
+                    response_first_token["token_top_logprobs"],
+                )
+                first_tokens = {
+                    logprob["token"] for logprob in top_five_logprobs
+                }
+                if "A" in first_tokens and "B" in first_tokens:
```

Comment:
This was not true for 2 / 5 when I tested it
```
Token 'A' or 'B' not found in [{'token': 'B', 'logprob': -1.0718586}, {'token': 'Verd', 'logprob': -1.0718586}, {'token': 'Based', 'logprob': -2.3218586}, {'token': '**', 'logprob': -2.5718586}, {'token': '"B', 'logprob': -2.5718586}]
Token 'A' or 'B' not found in [{'token': 'Verd', 'logprob': -1.0696266}, {'token': 'Based', 'logprob': -1.5696266}, {'token': 'To', 'logprob': -1.8196266}, {'token': 'The', 'logprob': -2.5696266}, {'token': 'A', 'logprob': -3.0696266}]
```

Maybe we can just add a print statement when this is not the case, just so that users are aware of this as a potential case? E.g.
```
for i, response in responses:
    ...
    if "A" in first_tokens and "B" in first_tokens:
        ...
    else:
        print("Token 'A' or 'B' was not found for the {i}th simulated annotator")
```

Reply:
Added this message. 
This seems to be a model-dependent issue, After switching the model to GPT3.5 Turbo, all annotators returned true in 100 runs, while GPT4o failed in 41 out of 100 runs.

Reply:
hmm yeah that makes sense that some models will be more robust to this than others. We'll probably experiment more with this to make this more robust, but I think it's ok for now!